filebeat/input/kafka/input.go

```diff
@@ -0,0 +1,285 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package kafka
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/input"
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	err := input.Register("kafka", NewInput)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Input contains the input and its config
+type kafkaInput struct {
+	config        kafkaInputConfig
+	rawConfig     *common.Config // The Config given to NewInput
+	context       input.Context
+	outlet        channel.Outleter
+	consumerGroup sarama.ConsumerGroup
+	sessionState  *kafkaSessionState
+	log           *logp.Logger
+	runOnce       sync.Once
+}
```

Comment:
do we need all fields? E.g. I don't see rawConfig being used.

Reply:
No, it was used in an earlier iteration but can be removed now